kzg/src/scheme.rs

```diff
@@ -124,6 +128,80 @@ impl KzgScheme {
         let pairing2 = Bls12_381::pairing(b, g2);
         pairing1 == pairing2
     }
+
+    /// Aggregates multiple commitments into one commitment using a random challenge
+    ///
+    /// For some challenge `v`, a list of commitments `(c_0, c_1, ... c_n) `
+    ///
+    /// We compute the aggregate commitment as `v^0 * c_0 + v^1 * c_1 + ...+ v^n * c_n`
+    ///
+    /// # Arguments
+    ///
+    /// * `commitments`: The commitments to be aggregated
+    /// * `challenge`: The random challenge
+    ///
+    /// # Returns
+    ///
+    /// Aggregated commitment from input commitments
+    pub fn aggregate_commitments(
+        commitments: &Vec<&KzgCommitment>,
+        challenge: &Fr,
+    ) -> KzgCommitment {
+        let mut pow = Fr::one();
+        let mut result = G1Projective::zero();
+
+        for commitment in commitments {
+            result += &commitment.0.mul(pow);
+            pow *= challenge;
+        }
+
+        KzgCommitment(result.into_affine())
+    }
+
+    /// Verifies that each proof is a valid proof of evaluation for `commitment_i` at `point_i`.
+    ///
+    /// This function is implemented according to the protocol on page 13 of the Plonk paper,
+    ///
+    /// # Arguments
+    ///
+    /// * `commitments`: The polynomials' commitments that need to be verified
+    /// * `points`: The corresponding challenge points
+    /// * `openings`: The corresponding open proof for polynomials
+    ///
+    /// # Returns
+    ///
+    /// `true` if all proofs is valid, otherwise `false`.
+    pub fn batch_verify(
+        &self,
+        commitments: &[KzgCommitment],
+        points: &[Fr],
+        openings: &[KzgOpening],
+    ) -> bool {
+        assert_eq!(commitments.len(), points.len());
+        assert_eq!(openings.len(), points.len());
+
+        let g1 = G1Point::generator();
+        let mut rng = StdRng::from_entropy();
+        let mut e_1 = G1Point::zero();
+        let mut e_2 = G1Point::zero();
+
+        for ((cm, z), KzgOpening(w, s)) in commitments.iter().zip(points).zip(openings) {
```

Comment:
can you write doc for this part, it's really hard for my brain to interpret what cm, z, w and s is every step